server/libs/domain/src/media/media.service.ts

```diff
@@ -127,20 +127,22 @@ export class MediaService {
       const output = join(outputFolder, `${asset.id}.mp4`);
       this.storageRepository.mkdirSync(outputFolder);
 
-      const { streams } = await this.mediaRepository.probe(input);
-      const stream = await this.getLongestStream(streams);
-      if (!stream) {
+      const { videoStreams, audioStreams, format } = await this.mediaRepository.probe(input);
+      const mainVideoStream = await this.getMainVideoStream(videoStreams);
+      const mainAudioStream = await this.getMainAudioStream(audioStreams);
+      const containerExtension = format.formatName;
+      if (!mainVideoStream || !mainAudioStream || !containerExtension) {
```

Comment:
This causes audio or video-only files to never be transcoded. Is that intended behavior? Also unneeded awaits for selecting the streams

Reply:
Not sure why my IDE wasn't complaining about the awaits, will remove those. We don't support audio only files in Immich, however video files without audio I guess could be an edge case we could consider.